File: source/livana-fe/src/types/response/adminUserResponse.ts
export interface AdminUserResponse {
  id: string;
  username: string;
  email: string;
  fullName: string | null;
  avatarUrl: string | null;
  enabled: boolean;
  status: UserStatus;
  roles: string[];
  lastLoginAt: string | null;
  createdAt: string;
}

export type UserStatus =
  | "ACTIVE"
  | "INACTIVE"
  | "BANNED"
  | "PENDING_VERIFICATION";

export type UserRole = "ROLE_ADMIN" | "ROLE_USER" | "ROLE_MODERATOR";

export interface PageResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  first: boolean;
  last: boolean;
  empty: boolean;
}

File: source/livana-be/src/main/java/octguy/livanabe/dto/response/AdminUserResponse.java
package octguy.livanabe.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import octguy.livanabe.enums.UserStatus;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminUserResponse {
    private UUID id;
    private String username;
    private String email;
    private String fullName;
    private String avatarUrl;
    private boolean enabled;
    private UserStatus status;
    private List<String> roles;
    private LocalDateTime lastLoginAt;
    private LocalDateTime createdAt;
}
